src/client/Microsoft.Identity.Client/Extensibility/AcquireTokenForClientBuilderExtensions.cs

```diff
@@ -11,6 +13,34 @@ namespace Microsoft.Identity.Client.Extensibility
     /// </summary>
     public static class AcquireTokenForClientBuilderExtensions
     {
+        /// <summary>
+        /// Specifies additional cache key components to use when caching and retrieving tokens.
+        /// </summary>
+        /// <param name="cacheKeyComponents">The list of additional cache key components.</param>
+        /// <param name="builder"></param>
+        /// <returns>The builder.</returns>
+        /// <remarks>
+        /// <list type="bullet">
+        /// <item><description>This api can be used to associate certificate key identifiers along with other keys with a particular token.</description></item>
+        /// <item><description>In order for the tokens to be successfully retrieved from the cache, all components used to cache the token must be provided.</description></item>
+        /// </list>
+        /// </remarks>
+        internal static AcquireTokenForClientParameterBuilder WithAdditionalCacheKeyComponents(this AcquireTokenForClientParameterBuilder builder,
+            IDictionary<string, string> cacheKeyComponents)
+        {
+            builder.ValidateUseOfExperimentalFeature();
+
+            if (cacheKeyComponents == null || cacheKeyComponents.Count == 0)
+            {
+                //no-op
+                return builder;
+            }
+
+            builder.CommonParameters.CacheKeyComponents = new SortedList<string, string>(cacheKeyComponents);
```

Comment:
Can you pls make this additive? In case 2 MISE extensions use this. i.e. if CacheKeyComponents  already exists, the entries.